src/Conjure/Language/Type.hs

```diff
@@ -239,3 +243,47 @@ typeCanIndexMatrix TypeBool{} = True
 typeCanIndexMatrix TypeInt {} = True
 typeCanIndexMatrix TypeEnum{} = True
 typeCanIndexMatrix _          = False
+
+morphing :: (?typeCheckerMode :: TypeCheckerMode)
```

Comment:
Can you write a one line description of this function please? Not sure I understand what it achieves.

Reply:
It is just a helper for extracting the type that a homomorphism is "morphing". Right now it is mostly a placeholder - there will be cases for Permutation and Sequence once they are supported.